src/test/recovery/Makefile

```diff
@@ -12,6 +12,22 @@ subdir = src/test/recovery
 top_builddir = ../../..
 include $(top_builddir)/Makefile.global
 
+# In PG15, Perl test modules have been moved to a new namespace
+# new() and get_new_node() methods have been unified to 1 method: new()
+# Relevant PG commits 201a76183e2056c2217129e12d68c25ec9c559c8
+#                     b3b4d8e68ae83f432f43f035c7eb481ef93e1583
+pg_version = $(shell $(PG_CONFIG) --version 2>/dev/null)
+pg_whole_version = $(shell echo "$(pg_version)"| sed -e 's/^PostgreSQL \([0-9]*\)\(\.[0-9]*\)\{0,1\}\(.*\)/\1\2/')
+pg_major_version = $(shell echo "$(pg_whole_version)"| sed -e 's/^\([0-9]\{2\}\)\(.*\)/\1/')
+
+ifeq ($(pg_major_version),13)
```

Comment:
Hmm, this works just because we have a single test file. If we had more than 1, we'd not be running this. (same for tap tests)

Can you see another way for doing this such that we wouldn't prevent future test additions?

Reply:
Ideally, I'd want to add the version check inside the test, but I didn't find a way of doing that. I could create two different folders. Any new test we add would need to include the Postgres node differently for different PG versions anyway.